src/api/middleware/errorHandler.js

```diff
@@ -32,10 +34,23 @@ const handler = () => (err, req, res, next) => {
             errors.push({ message, path, type });
         }
 
-        if (errors.length !== 0)
-            result.errors = errors;
+        if (errors.length !== 0) result.errors = errors;
     }
     return res.status(400).send(result);
 };
 
-module.exports = { handler, ERROR_TYPES };
+const dbHandler = () => (err, req, res, next) => {
+    if (!err.hasOwnProperty("name") || err.name !== "MongoError") {
+        return next(err);
+    }
+
+    const result = {
+        success: false,
+        error_code: ERROR_TYPES.DB_ERROR,
+        error: errorExtractor(err),
+    };
+
+    return res.status(500).send(result);
```

Comment:
Currently, in the only case that this happens the status should probably be 400. But we should revisit this later, the status should be calculated from some more context, such as a `req.something` added by the request handler to give context to the db error handler.